### PR TITLE
Completed CMS story to be merged to master

### PR DESCRIPTION
**Pull request by katestarks**

### CODE REVIEW

File: cms_index.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+if(isset($_POST['deleteAboutMeText'])) {
+    $deleteAboutMePara = $_POST['deleteAboutMeText'];
+    $aboutMeTextDeleted = deleteAboutMeText($db, $deleteAboutMePara);
+    $deleteSuccessMessage = deleteAboutMeSuccess($aboutMeTextDeleted);
+}
+
+$aboutMeTextAndQuote = getAboutMeTextAndQuote($db);
+$displayAboutMeDropdown = aboutMeTextDropdown($aboutMeTextAndQuote);
+
+
```
Comment (charliecog) [inline cms_index.php]: remove line break

File: cms_index.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Admin for Kate's Portfolio Page</title>
+    <link rel='stylesheet' type='text/css' href='css/normalize.css'/>
+    <link href="https://fonts.googleapis.com/css?family=Quicksand:300,500" rel="stylesheet">
+    <link rel='stylesheet' type='text/css' href='css/cms_styles.css'/>
+</head>
+<body>
+<h1>Content Managing Kate's Portfolio Page</h1>
```
Comment (charliecog) [inline cms_index.php]: indent by one tab

File: cms_index.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+<section class="aboutMe">
+    <h2>About Me</h2>
+    <form method="POST" action="" id="addAboutMeText">
```
Comment (charliecog) [inline cms_index.php]: change action to cms_index.php so it looks intentional that you are submitting to the same page

File: cms_index.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
+<?php
+require_once 'php/dbConnection.php';
+require_once 'php/functions.php';
+
+$db = getdbConnection();
```
Comment (charliecog) [inline cms_index.php]: getdbConnection should be camel case getDbConnection

File: cms_index.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            <textarea class="typeText" name="editAboutMeTextAndQuote" form="editAboutMeTextAndQuote">
+<?php
+            if (isset($displayAboutTextToEdit)) {
+                echo $displayAboutTextToEdit;
+            }?></textarea>
+                <?php
+                if (isset($editId)) {
+                    echo '<input type="hidden" value=' . $editId . ' name="editId">';
+                    }
+                echo $displaySubmitEditButton;
```
Comment (charliecog) [inline cms_index.php]: line 84 + 85 need isset around them to prevent chance of an undefined index error

File: php/functions.php [hunk 6]
```diff
@@ -0,0 +1,250 @@
+<?php
+
+require_once "dbConnection.php";
```
Comment (charliecog) [inline php/functions.php]: dont need that here

File: php/functions.php [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+/**
+ * Retrieves content from the database excluding line 1
+ *
+ * @param PDO $db database of about me text
+ *
+ * returns an array of content field excluding the first line of the database
```
Comment (charliecog) [inline php/functions.php]: @return for your docblocking

File: php/functions.php [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * Displays data held in the array fetched from the database limited to line 1
+ *
+ * @param array $aboutMeTexts result of function to get the about me text (limited to line 1) assigned to a variable
+ *
+ * returns string of the array items concatenated with html
```
Comment (charliecog) [inline php/functions.php]: @return

File: php/functions.php [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+    }
+    return $result;
+}
+
+/**
+ * Retrieves content from the database limited to line one
+ *
+ * @param PDO $db database of about me text
+ *
+ * returns an array of content field limited to the first line of the database
```
Comment (charliecog) [inline php/functions.php]: @return

File: php/functions.php [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * Displays data held in the array fetched from the database which excludes line 1
+ *
+ * @param array $aboutMeTexts result of function to get the about me text (excl line 1) assigned to a variable
+ *
+ * returns string of the array items concatenated with html
```
Comment (charliecog) [inline php/functions.php]: @return

File: php/functions.php [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+ * Update data base with edited content
+ *
+ * @param PDO $db database connection
+ * @param bool returned from the function which checks whether there are characters in the string
+ * @param string $submitEditText edited content from the form
+ * @param string $editId database id saved in the session
+ *
+ * @return bool updates the database with edited content based on id
+ */
+function updateAboutMeQuoteAndText(PDO $db, bool $checkEditText, string $submitEditText, string $editId) :bool {
```
Comment (charliecog) [inline php/functions.php]: function should just do one thing so do the check on checkEditText outside the function elsewhere.

**Review by charliecog — COMMENTED** [review]
looks good, couple of changes and then you are there

File: cms_index.php (hunk 1 above)
Comment (katestarks) [reply cms_index.php]: Done

File: cms_index.php (hunk 2 above)
Comment (katestarks) [reply cms_index.php]: Done

File: cms_index.php (hunk 3 above)
Comment (katestarks) [reply cms_index.php]: Done

File: cms_index.php (hunk 4 above)
Comment (katestarks) [reply cms_index.php]: Done

File: cms_index.php (hunk 5 above)
Comment (katestarks) [reply cms_index.php]: Done

File: php/functions.php (hunk 6 above)
Comment (katestarks) [reply php/functions.php]: Done

File: php/functions.php (hunk 7 above)
Comment (katestarks) [reply php/functions.php]: Done

File: php/functions.php (hunk 8 above)
Comment (katestarks) [reply php/functions.php]: Done

File: php/functions.php (hunk 9 above)
Comment (katestarks) [reply php/functions.php]: Done

File: php/functions.php (hunk 10 above)
Comment (katestarks) [reply php/functions.php]: Done

File: php/functions.php (hunk 11 above)
Comment (katestarks) [reply php/functions.php]: Done

File: cms_index.php [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+            <select class="aboutMeDropdown" name="selectAboutMeText">
+                <?php
+                echo $displayAboutMeDropdown;
+                ?>
+            </select>
+            <input type="submit" value="select text">
+        </form>
+        <form method="POST" action="" id="editAboutMeTextAndQuote">
+                <textarea class="typeText" name="editAboutMeTextAndQuote" form="editAboutMeTextAndQuote">
+    <?php
```
Comment (charliecog) [inline cms_index.php]: remember to put the php tag right next to the edge of the textarea on the line above to get rid of the space in your form

**Review by charliecog — APPROVED** [review]
looks good!

**Pull request merged**